File: smart-university-frontend/src/types/reclamation.ts
export type ReclamationStatus = 'PENDING' | 'RESOLVED' | 'REJECTED';

export interface Reclamation {
  id: number;
  studentName: string;
  subject: string;
  description: string;
  type: string;
  status: ReclamationStatus;
  createdAt: string;
  updatedAt: string;
}

export interface CreateReclamationDTO {
  studentName: string;
  subject: string;
  description: string;
  type: string;
  status?: string;
}

File: reclamation_service/src/main/java/org/example/reclamationservice/entity/Reclamation.java
package org.example.reclamationservice.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Reclamation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String studentName;

    private String subject;

    @Column(length = 2000)
    private String description;

    private String type;

    private String status;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}